src/Datadog.Trace/Plugins/PluginManager.cs

```diff
@@ -0,0 +1,99 @@
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Reflection;
+using Datadog.Trace.Configuration;
+using Datadog.Trace.Logging;
+using Datadog.Trace.Util;
+using Datadog.Trace.Vendors.Newtonsoft.Json.Linq;
+
+namespace Datadog.Trace.Plugins
+{
+    internal static class PluginManager
+    {
+        private static readonly IDatadogLogger Log = DatadogLogging.GetLoggerFor(typeof(PluginManager));
+
+        internal static IReadOnlyCollection<IOTelExtension> TryLoadPlugins(JsonConfigurationSource pluginsConfig)
+        {
+            if (pluginsConfig == null)
+            {
+                return ArrayHelper.Empty<IOTelExtension>();
+            }
+
+            var targetFramework = FrameworkDescription.Instance.TargetFramework;
+
+            Log.Debug("Executing plugins configuration: {0}", pluginsConfig);
+            Log.Information("Trying to load plugins with target framework '{0}'.", targetFramework);
+
+            // TODO: Detect if objects instead of path
```

Comment:
I do not understand this comment 😄

Reply:
https://github.com/open-telemetry/opentelemetry-dotnet-instrumentation/pull/129#discussion_r616554758
That's for the future, where we discussed about security / integrity of the loaded plugin.

Reply:
Changed a bit, should be less cryptic 😄